### ISSUE TITLE
Option to store scoped packages in Artifactory layout

### ISSUE THREAD

**Issue by roarbr** [larsw/npm-leech#13]
Hi,
In he npmjs.org registry scoped packages has a tarball path like this:
```
# npm info '@types/qs@latest' --json dist.tarball
"https://registry.npmjs.org/@types/qs/-/qs-6.9.6.tgz"
```
I.e. "@scope/packagename/-/packagename-version.tgz"
When downloaded by npm-leech, the file structure in the tar file is:
```
# tar tvf npm-tarballs.tar 
-rw-r--r-- 0/0            2405 2021-05-05 18:23 @types/qs-6.9.6.tgz
```
When this tar-file is imported to Artifactory using the web-interface, this layout from the tar file is kept in Artifactory. This is not the "correct" path inside Artifactory, it uses a path like this for scoped packages:
```
@scope/packagename/~/@scope/packagename-version.tgz
```
When importing the tar file from npm-leech into Artifactory, the incorrect layout will cause issues with scoped packages that also exist without a scope with the same name and version. Like "@types/qs" and "qs". It will cause checksum error when trying to install (npm install) a package like this using Artifactory as a registry. See this bug for more info:
https://www.jfrog.com/jira/browse/RTFACT-7668

Several issues like this is reported for Artifactory, but no response from the developer that I could see.

Pushing the packages to Artifactory using "npm publish" will go through the Artifactory API and store the file with the layout expected by Artifactory.

But I hoped this could be fixed in npm-leech by adding a option to store in Artifactory layout inside the tar file. This way the bulk import of the tar file from Artifactory web interface would work out of the box.

**Comment by larsw** [comment]
I accept Pull Requests! :-)

**Comment by roarbr** [comment]
Pull request added:
https://github.com/larsw/npm-leech/pull/14

New to pull requests on GH, the linking between issue and pullreq is not available, can't make that link.

I have newer coded in node/js before, so have that in mind while reviewing the code :)

**Comment by larsw** [comment]
Solved via #14
